sfall/Modules/Explosions.cpp

```diff
@@ -180,27 +191,33 @@ int _stdcall ExplosionsMetaruleFunc(int mode, int arg1, int arg2) {
 				SafeWrite8(0x411B54, 6); // last direction
 			}
 			break;
-		case EXPL_FORCE_EXPLOSION_ART: {
+		case EXPL_FORCE_EXPLOSION_ART:
 			for (int i = 0; i < numArtChecks; i++) {
 				SafeWrite32(explosion_art_adr[i], (BYTE)arg1);
 			}
 			break;
-		}
 		case EXPL_FORCE_EXPLOSION_RADIUS:
-			SafeWrite32(explosion_radius_grenade, arg1);
-			SafeWrite32(explosion_radius_rocket, arg1);
+			SetExplosionRadius(arg1, arg1);
 			break;
-		case EXPL_FORCE_EXPLOSION_DMGTYPE: {
+		case EXPL_FORCE_EXPLOSION_DMGTYPE:
 			for (int i = 0; i < numDmgChecks; i++) {
 				SafeWrite8(explosion_dmg_check_adr[i], (BYTE)arg1);
 			}
 			break;
-		}
+		case EXPL_STATIC_EXPLOSION_RADIUS:
+			if (arg1 > 0) set_expl_radius_grenade = arg1;
+			if (arg2 > 0) set_expl_radius_rocket = arg2;
+			SetExplosionRadius(set_expl_radius_grenade, set_expl_radius_rocket);
+			break;
+		default:
+			return -1;
```

Comment:
Consider printing error to the debug log while returning -1. I think some of the other functions had those.